enhancements/operator-dependency-resolution.md

```diff
@@ -0,0 +1,222 @@
+
+There was also some initial discussion around what the user workflow would be to edit/override dependencies. Given the current state of the APIs that exist (specifically the Subscription and InstallPlan APIs), those would require extensive changes to create a trivial override for dependencies. Even then, there is a configurability problem where a single install overriding a dependency may not enough and a cluster admin may want to create a cluster wide or namespaced override that is persistent and editable. Initially, there was some discussion about the possibility of a dependency Override API or the possibility of using the Operator API for this purpose, but those discussions are ultimately outside the scope of the immediate (3+ month) timeframe and scope of this enhancement proposal, and will be taken up further in the future.
+
+### User Stories
+
+#### Explicit Operator Dependency Metadata
+
+As an operator author, I want to specify dependencies explicitly in the bundle. For example, I would like to specify that my operator depends on etcd operator version 0.9.0.
+* Explicit operator dependency is natural, intuitive and safe to understand
+* Avoid unexpected situation where two different operators owning the same CRD/API
+
+Acceptance Criteria:
+I can specify explicit dependencies in a file in the bundle and it can be included with other manifests and metadata in the bundle image.
+
+Internal details:
+Currently, the plan is to have `dependencies.yaml` as a centralized location for explicit dependency specification. However, we may plan to include that information in Dockerfile/manifest list so that information can be query without unpack the bundle image to see its content.
+
+#### Explicit Operator Dependency Validation
+
+As an operator author, I want to validate dependencies solution in the bundle. For example, I would like to make sure OLM will resolve and install all the dependencies I specify in the bundle correctly.
+* Allow operator authors to test dependency resolution for their operators before publishing
+
+Acceptance Criteria:
+I can use some toolings to list the operators that will be installed for my operator that will meet all of my dependency requirements that I specify in the bundle.
+
+Internal details:
+The proposal sat solver (resolver) will be in a separate repository in operator-framework and it can be imported into operator-registry and `opm` can use the resolver to perform the dependency resolution that OLM will do. Then, operator author can utilize the `opm` binary for testing and validation.
+
+#### Cluster Admin Dependency Observability
+
+As a cluster admin, I want to see the list of dependencies that are going to be installed when I inspect an operator.
+* Allow cluster admin to identify which operators that are going to be installed in order to satisfy the dependency requirement for a particular operator
+
+Acceptance Criteria:
+The cluster admin can see the dependency list in InstallPlan in the UI.
```

Comment:
Will this include nested dependencies? e.g MyApp depends on istio which depends on etcd

Reply:
That's the plan